### PR TITLE
FilterSelectionDialog: handle non-removable fields

### PR DESCRIPTION
**Pull request by labkey-nicka**
#### Rationale
Update `FieldSelectionDialog` test component to handle removing all selected fields edge case where some fields are not actually removable.

#### Related Pull Requests
- https://github.com/LabKey/limsModules/pull/971

#### Changes
- Skip all fields removed check if not all fields can be removed.

### CODE REVIEW

File: src/org/labkey/test/components/ui/grids/FieldSelectionDialog.java [hunk 1]
```diff
@@ -386,20 +386,25 @@ public FieldSelectionDialog removeFieldFromSelected(String field, int index)
     public FieldSelectionDialog removeAllSelectedFields()
     {
         List<WebElement> allItems = elementCache().getListItemElements(elementCache().selectedFieldsPanel);
+        boolean removedAll = true;
 
-        for(WebElement listItem : allItems)
+        for (WebElement listItem : allItems)
```
Comment (labkey-chrisj) [inline src/org/labkey/test/components/ui/grids/FieldSelectionDialog.java:391]: Occasionally this method will fail because not every item was removed-- one theory I have is that it might be due to how they slide up when the previous one is removed.  It might be more reliable to iterate from the bottom up, if it's possible for the click to 'miss' if it happens when the target is moving

**Review by labkey-chrisj — APPROVED** [review]

File: src/org/labkey/test/components/ui/grids/FieldSelectionDialog.java (hunk 1 above)
Comment (labkey-nicka) [reply src/org/labkey/test/components/ui/grids/FieldSelectionDialog.java:391]: I'm going to leave as-is since I'm not looking to triage intermittent failures with this work. What you are proposing seems reasonable.
